sdk/servicebus/azure-servicebus/samples/sync_samples/failure_and_recovery.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+An (overly) verbose sample demonstrating possible failure modes and potential recovery patterns.
+
+Many of these catches are present for illustrative or duplicate purposes, and could be condensed or elided
+in a production scenario depending on the system design.
+"""
+
+# pylint: disable=C0111
+
+import os
+from azure.servicebus import ServiceBusClient, ServiceBusMessage
+from azure.servicebus.exceptions import (
+    MessageSizeExceededError,
+    ServiceBusConnectionError,
+    ServiceBusAuthorizationError,
+    ServiceBusAuthenticationError,
+    OperationTimeoutError,
+    ServiceBusError,
+    ServiceBusCommunicationError,
+    MessageAlreadySettled,
+    MessageLockLostError,
+    MessageNotFoundError
+)
+
+CONNECTION_STR = os.environ['SERVICE_BUS_CONNECTION_STR']
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+def send_batch_messages(sender):
+    batch_message = sender.create_message_batch()
+    for i in range(10):
+        try:
+            batch_message.add_message(ServiceBusMessage("Data {}".format(i)))
+        except MessageSizeExceededError:
+            # The body provided to the ServiceBusMessage is too large.
+            # This must be handled at the application layer, by breaking up or condensing.
+            continue
+        except ValueError:
+            # ServiceBusMessageBatch object reaches max_size.
+            # New ServiceBusMessageBatch object can be created here to send more data.
+            break
```

Comment:
we only raise MessageSizeExceededError when `add_message` exceeds the size (although it inherits from ValueError).
this except part is redundant or just showing that user could catch ValueError?

Reply:
Hah, I had cargo culted that from the send_queue sample, and then checked the "failed" code in the docstring so I unioned the two rather than realizing it was the same thing.  However, your thought about "should you just catch valueerror" makes me realize we elided the "improper payload" failure case, so I'll split this into two pieces as I fix it.